File: medplum-java/medplum-model/src/main/java/com/fhir/Researchelementdefinition_characteristicStudyeffectivegroupmeasure.java
package com.fhir;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Locale;
import javax.annotation.Nonnull;

public final class Researchelementdefinition_characteristicStudyeffectivegroupmeasure {
  public static final Researchelementdefinition_characteristicStudyeffectivegroupmeasure MEDIAN = new Researchelementdefinition_characteristicStudyeffectivegroupmeasure(Value.MEDIAN, "MEDIAN");

  public static final Researchelementdefinition_characteristicStudyeffectivegroupmeasure MEAN = new Researchelementdefinition_characteristicStudyeffectivegroupmeasure(Value.MEAN, "MEAN");

  private final Value value;

  private final String string;

  Researchelementdefinition_characteristicStudyeffectivegroupmeasure(Value value, String string) {
    this.value = value;
    this.string = string;
  }

  public Value getEnumValue() {
    return value;
  }

  @Override
  @JsonValue
  public String toString() {
    return this.string;
  }

  @Override
  public boolean equals(Object other) {
    return (this == other) 
      || (other instanceof Researchelementdefinition_characteristicStudyeffectivegroupmeasure && this.string.equals(((Researchelementdefinition_characteristicStudyeffectivegroupmeasure) other).string));
  }

  @Override
  public int hashCode() {
    return this.string.hashCode();
  }

  public <T> T accept(Visitor<T> visitor) {
    switch (value) {
      case MEDIAN:
        return visitor.visitMEDIAN();
      case MEAN:
        return visitor.visitMEAN();
      case UNKNOWN:
      default:
        return visitor.visitUnknown(string);
    }
  }

  @JsonCreator(
      mode = JsonCreator.Mode.DELEGATING
  )
  public static Researchelementdefinition_characteristicStudyeffectivegroupmeasure valueOf(
      @Nonnull String value) {
    String upperCasedValue = value.toUpperCase(Locale.ROOT);
    switch (upperCasedValue) {
      case "MEDIAN":
        return MEDIAN;
      case "MEAN":
        return MEAN;
      default:
        return new Researchelementdefinition_characteristicStudyeffectivegroupmeasure(Value.UNKNOWN, upperCasedValue);
    }
  }

  public enum Value {
    MEAN,

    MEDIAN,

    UNKNOWN
  }

  public interface Visitor<T> {
    T visitMEAN();

    T visitMEDIAN();

    T visitUnknown(String unknownType);
  }
}

File: medplum-typescript/src/model/fhir/Researchelementdefinition_characteristicStudyeffectivegroupmeasure.ts
export type Researchelementdefinition_characteristicStudyeffectivegroupmeasure =
    | Researchelementdefinition_characteristicStudyeffectivegroupmeasure.Mean
    | Researchelementdefinition_characteristicStudyeffectivegroupmeasure.Median;

export const Researchelementdefinition_characteristicStudyeffectivegroupmeasure = {
    Mean: "mean" as Researchelementdefinition_characteristicStudyeffectivegroupmeasure.Mean,
    Median: "median" as Researchelementdefinition_characteristicStudyeffectivegroupmeasure.Median,

    _visit: <Result>(value: Researchelementdefinition_characteristicStudyeffectivegroupmeasure, visitor: Researchelementdefinition_characteristicStudyeffectivegroupmeasure._Visitor<Result>): Result => {
        switch (value) {
            case Researchelementdefinition_characteristicStudyeffectivegroupmeasure.Mean: return visitor.mean();
            case Researchelementdefinition_characteristicStudyeffectivegroupmeasure.Median: return visitor.median();
            default: return visitor._unknown();
        }
    },
};

export declare namespace Researchelementdefinition_characteristicStudyeffectivegroupmeasure {
    export type Mean = "mean" & {
        __Mean: void,
    };
    export type Median = "median" & {
        __Median: void,
    };

    export interface _Visitor<Result> {
        mean: () => Result;
        median: () => Result;
        _unknown: () => Result;
    }
}
